packages/dataviews/src/types.ts

```diff
@@ -53,6 +53,26 @@ export type ValidationContext = {
 	elements?: Option[];
 };
 
+/**
+ * An abstract interface for Field based on the field type.
+ */
+export type FieldTypeDefinition< Item > = {
```

Comment:
👍

One thing I'm wondering is that "field types" shouldn't depend on "Item" they should work for any item value of the defined type. So I think we have somethings wrong in the abstraction. `sort, isValid and Edit` should probably not receive `items` at all and just receive values. But it's also something that should be solved in a separate PR.